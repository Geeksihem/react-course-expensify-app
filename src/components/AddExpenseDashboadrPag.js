import React from 'react'; 
import ExpenseForm from './ExpenseForm'; 
import {connect} from 'react-redux'; 
import {addExpense} from '../actions/expenses';  
export class AddExpenseDashboadrPag extends  React.Component {
    onSubmit =  (expense)=>{
        //props.dispatch(addExpense(expense ));
       this.props.onSubmit(expense); 
       this.props.history.push('/');  
    }; 
     render(){
        return(
            <div> 
            add page 
            <ExpenseForm
             onSubmit =  {this.onSubmit}
            /> 
            </div>
        )
    } 
   }  
    const mapDispatchToProps = (dispatch)=>({
        onSubmit: (expense)=> dispatch(addExpense(expense))
       
    } )
export default  connect( undefined, mapDispatchToProps)(AddExpenseDashboadrPag); 