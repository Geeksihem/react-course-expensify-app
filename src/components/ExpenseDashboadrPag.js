import React from 'react'; 
import ConnectedExpenseList from './ExpenseList'; 
import ExpenseListFilters from './ExpenseListFilters'; 

const ExpenseDashboadrPag = ()=>( 
    <div> 
    <ExpenseListFilters /> 
<ConnectedExpenseList  />
    </div>
    ) ; 
 export default ExpenseDashboadrPag; 