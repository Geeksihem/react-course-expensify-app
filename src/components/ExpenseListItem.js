import React from 'react'; 
import {Link} from 'react-router-dom'; 
 
// export a staless funtion 
const ExpenseListItem  = ({  dispatch,  id, description,  amount, createdAt, note } )=>(
    <div> 
<Link to={`/edit/${id}`}>  <p> dddd</p>
</Link>
    <p> {amount} -- {createdAt} </p> 
    <p> {note}</p>
 
    </div>
)

export default  ExpenseListItem   ; 
 