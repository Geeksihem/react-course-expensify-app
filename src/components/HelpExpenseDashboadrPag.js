import React from 'react'; 
       
const HelpExpenseDashboadrPag = ()=>( 
    <div> 
    help  page 
    </div>
    ) ; 
export default HelpExpenseDashboadrPag;     