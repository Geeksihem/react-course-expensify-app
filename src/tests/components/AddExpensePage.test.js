import React from 'react'; 
import {shallow } from 'enzyme'; 
import {AddExpenseDashboadrPag} from '../../components/AddExpenseDashboadrPag'; 
import  expenses from '../fixture/expenses';

let history, onSubmit, wrapper; 

beforeEach(()=>{
      onSubmit = jest.fn(); 
      history = {push: jest.fn()}; 
      wrapper = shallow(<AddExpenseDashboadrPag onSubmit= {onSubmit} history= {history} />)
 }); 
test('should render  addexpense paage ', ()=>{
   expect(wrapper).toMatchSnapshot(); 
}) 
test('should handle onSubmit ', ()=>{ 
  wrapper.find('ExpenseForm').prop('onSubmit')(expect[1]); 
   expect(history.push).toHaveBeenLastCalledWith('/'); 
   expect(onSubmit).toHaveBeenLastCalledWith( expect[1]); 
 })