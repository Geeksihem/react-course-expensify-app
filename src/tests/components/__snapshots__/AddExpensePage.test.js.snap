// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render  addexpense paage  1`] = `
<div>
  add page
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
